### PR TITLE
Add sentinel to terraform feature

### PR DESCRIPTION
**Pull request by sdrzazga**
Hi.

Based on my feature request #437 I'd like to contribute with this PR to the potential installation of sentinel with the terraform feature.
The description was copied from the corresponding [documentation](https://docs.hashicorp.com/sentinel).
Unfortunately, Hashicorp doesn't release sentinel via GH hence the introduction of a separate function in the `install.sh` to resolve the version from the release URL.

### CODE REVIEW

**Comment by sdrzazga** [comment]
@microsoft-github-policy-service agree

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Looks great, thank you! Left minor comments.

File: src/terraform/README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ Installs the Terraform CLI and optionally TFLint and Terragrunt. Auto-detects la
 | version | Terraform version | string | latest |
 | tflint | Tflint version | string | latest |
 | terragrunt | Terragrunt version | string | latest |
+| installSentinel | Install sentinel, a language and framework for policy built to be embedded in existing software to enable fine-grained, logic-based policy decisions. | boolean | false |
```
[the remark this answers is not in the document]
Comment (sdrzazga) [reply src/terraform/README.md:21]: Absolute...silly me 🙄

File: src/terraform/install.sh [hunk 2]
```diff
@@ -198,6 +224,31 @@ if [ "${TERRAGRUNT_VERSION}" != "none" ]; then
     mv -f /tmp/tf-downloads/${terragrunt_filename} /usr/local/bin/terragrunt
 fi
 
+if [ "${INSTALL_SENTINEL}" = "true" ]; then
+    SENTINEL_VERSION="latest"
+    find_sentinel_version_from_url SENTINEL_VERSION 'https://releases.hashicorp.com/sentinel'
+    sentinel_filename="sentinel_${SENTINEL_VERSION}_linux_${architecture}.zip"
+    sentinel_releases_url='https://releases.hashicorp.com/sentinel'
```
[the remark this answers is not in the document]
Comment (sdrzazga) [reply src/terraform/install.sh:231]: I agree, this happens when you do changes late in the day.

File: src/terraform/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdrzazga) [reply src/terraform/README.md]: File was reverted

File: src/terraform/devcontainer-feature.json [hunk 3]
```diff
@@ -33,6 +33,11 @@
             "default": "latest",
```
[the remark this answers is not in the document]
Comment (sdrzazga) [reply src/terraform/devcontainer-feature.json:33]: Raised version to 1.2.0

**Comment by sdrzazga** [comment]
@samruddhikhandale thank you very much for the swift review and comments.
I promise to do better in the future and don't do late-night commits anymore.

**Review by samruddhikhandale — APPROVED** [review]
Awesome, thanks for contributing! ⚡

**Pull request merged**